src/api/app/models/project.rb

```diff
@@ -31,6 +31,7 @@ class ForbiddenError < APIException
   end
 
   before_destroy :cleanup_before_destroy
+  after_destroy :delete_on_backend
```

Comment:
On the other hand the `delete_on_backend` rescues any exception related to backend communications...